### PR TITLE
Dashboard config

### LINKED ISSUE #22: Subscribe Script
### Steps

- [x] Implement a form in the Streamlit dashboard for users to input product URLs.
- [x] Validate the product URL and fetch initial product details.
- [x] Save the subscription details (including user ID and product ID) in the database.
- [x] Set up a confirmation mechanism to verify the subscription.
- [x] Integrate the subscribe script with the backend to start tracking the product.

### User Story

As a user, I want to subscribe to a product tracking service by entering the product URL, so I can monitor price changes.

### LINKED ISSUE #32: Add Product Entry
### Steps

- [x] Create a form in the Streamlit dashboard for entering product details (URL).
- [x] Validate the input data to ensure it is correct and complete.
- [x] Find the product code and name using the URL
- [x] Save the product details in the products table in the database.
- [x] Link the product to the user's tracking list with an optional price threshold.
- [x] Integrate with the backend to start monitoring the new product entry.

### User Story

As a user, I want to add a product to my tracking list by entering the product details, so I can monitor its price changes.

### LINKED ISSUE #33: View Price History
### Steps

- [x] Retrieve historical price data from the database.
- [x] Create visualisations (graphs, charts) to display the price trends.
- [x] Integrate these visualisations into the Streamlit dashboard.
- [x] Ensure the data is presented in a user-friendly and informative manner.

### User Story

As a user, I want to view the price history of my tracked products to analyse price trends over time.

### LINKED ISSUE #34: Remove Product from Tracking
### Steps

- [x] Implement functionality in the Streamlit dashboard to view tracked products.
- [x] Develop a feature to remove a product from the tracking list.
- [x] Update the database to reflect the removal of the tracking request.

### User Story

As a user, I want to remove a product from my tracking list so that I no longer receive updates about it.

### LINKED ISSUE #35: Subscribe/Add url
### Steps

- [x] Integrate the subscribe and add product entry scripts into a unified process.
- [x] Ensure smooth user flow from entering the product URL to subscription confirmation.
- [x] Implement error handling and user feedback mechanisms.

### User Story

As a user, I want a seamless process to subscribe and add products to my tracking list so that I can quickly set up my monitoring preferences.

### LINKED ISSUE #123: Update Dashboard Configuration
**Todo**
- [x] Update the dashboard configuration to follow company theme.
- [x] Add extra styling by manually injecting CSS and HTML.

**Story**
By being able to implement this styling, we can provide the user with a more enriching experience that better represents our brand.

### PR DESCRIPTION
**Pull request by BerkayDur**
This PR contains the code that finalises the design decisions on the dashboard: closes #123 

It also closes Issues which are loose ends: closes #35, closes #34, closes #33, closes #32, closes #22.

### CODE REVIEW

File: Dashboard/pages/price.py [hunk 1]
```diff
@@ -29,24 +29,29 @@ def get_encode_price_reading(
...
         threshold_line_enc = threshold_chart.encode(
-            x=alt.X("reading_at:T", title="Taken at"),
+            x=alt.X("reading_at:T", title="Time"),
             y=alt.Y("price", title = "Price (£)"),
             color=alt.value("#FF0000"),
+            text='text'
         )
-        return threshold_line_enc.mark_line() + chart.mark_point() + chart.mark_line()
-    return chart.mark_point() + chart.mark_line()
+        return (threshold_line_enc.mark_text(align='right', baseline='middle', dx=75, dy=0).transform_filter(alt.datum.text != None)
```
Comment (A-Faris) [inline Dashboard/pages/price.py]:
```suggestion
        return (threshold_line_enc.mark_text(align='right', baseline='middle',
        dx=75, dy=0).transform_filter(alt.datum.text is not None)
```

File: Dashboard/pages/add_product.py [hunk 2]
```diff
@@ -45,16 +45,19 @@ def subscribe_to_product(conn: connection, product_url: str, price_threshold: bo
...
     with st.container(border=True,):
         with st.form("subscribe_to_product", clear_on_submit=True, border=False):
-            product_url = st.text_input("Enter the product URL:")
+            product_url = st.text_input("**Paste the product link:**")
+            st.write('')
             price_threshold = st.text_input(
-                "Enter a threshold (£):", placeholder="Can be left empty")
-            if st.form_submit_button("Track Product", type="primary"):
+                "**Alert me when this product is below:** **:orange[*]**", placeholder="£")
+            st.markdown('<span style="font-size:0.8rem; position:relative; top:-1rem;">**:orange[*]** Leave this empty to be alerted on all price decreases</span>', unsafe_allow_html=True)
```
Comment (A-Faris) [inline Dashboard/pages/add_product.py:56]:
```suggestion
            st.markdown('<span style="font-size:0.8rem; position:relative; top:-1rem;">**:orange[*]**
            Leave this empty to be alerted on all price decreases</span>', unsafe_allow_html=True)
```

File: Dashboard/Dockerfile [hunk 3]
```diff
@@ -7,14 +7,23 @@ RUN mkdir .streamlit
...
-COPY logo/hermes_logo_full.png logo
+COPY logo/asos_logo.png logo
+COPY logo/asos_logo.svg logo
+COPY logo/hermes_icon_crop.png logo
+COPY logo/hermes_icon.png logo
+COPY logo/hermes_logo_black.png logo
+COPY logo/hermes_logo_orange.png logo
 COPY logo/hermes_logo.png logo
+COPY logo/patagonia_bear.png logo
+COPY logo/patagonia_logo.png logo
```
Comment (A-Faris) [inline Dashboard/Dockerfile:18]: Remove the images that are not needed

File: Dashboard/navigation.py [hunk 4]
```diff
@@ -21,22 +21,23 @@ def get_current_page_name() -> str | None:
 def make_sidebar() -> None:
     """Creates sidebar"""
     page_name = get_current_page_name()
-
     with st.sidebar:
-        st.logo("logo/hermes_icon.png")
+        st.logo("logo/hermes_logo.png")
```
Comment (A-Faris) [inline Dashboard/navigation.py:25]: consider using the orange logo

File: Dashboard/pages/email_alerts.py [hunk 5]
```diff
@@ -14,9 +14,10 @@
 
 def email_alerts_page(boto_ses_client: ses_client):
     '''Contains the StreamLit email alerts page'''
-
-    st.title('Email Alerts')
-
+    st.markdown('<h2 class="pageTitle">Email Alerts</h2>', unsafe_allow_html=True)
+    st.write('')
+    st.markdown('<p style="font-size:1.3rem;"><span style="color:var(--orange);"><b>Subscribe</b></span> to stay <b>alerted</b> on price drops for your favourite products.</p>', unsafe_allow_html=True)
```
Comment (A-Faris) [inline Dashboard/pages/email_alerts.py:19]:
```suggestion
    st.markdown('<p style="font-size:1.3rem;"><span style="color:var(--orange);">
    <b>Subscribe</b></span> to stay <b>alerted</b> on price drops for your favourite products.</p>',
    unsafe_allow_html=True)
```

File: Dashboard/pages/price.py [hunk 6]
```diff
@@ -29,24 +29,29 @@ def get_encode_price_reading(
...
-    )
+        x=alt.X("reading_at:T", title="Time", axis=alt.Axis(labels=False)),
+        y=alt.Y("price", title="Price (£)", scale=alt.Scale(zero=False))
+        )
     if price_threshold is not None:
         max_reading = price_reading_at["reading_at"].max()
         min_reading = price_reading_at["reading_at"].min()
         price_threshold_df = pd.DataFrame([{"reading_at": min_reading, "price": price_threshold},
-                                           {"reading_at": max_reading, "price": price_threshold}])
+                                           {"reading_at": max_reading, "price": price_threshold, 'text': 'Alert Threshold'}])
```
Comment (A-Faris) [inline Dashboard/pages/price.py]:
```suggestion
                                           {"reading_at": max_reading, "price": price_threshold,
                                           'text': 'Alert Threshold'}])
```

File: Dashboard/pages/price.py [hunk 7]
```diff
@@ -62,16 +67,21 @@ def change_threshold_in_db(
 
 def change_threshold(conn: connection, product_information: dict) -> None:
     '''Contains the streamlit display logic for changing the threshold in the dashboard.'''
-    text_input_placeholder = None
+    text_input_placeholder = '£'
     if product_information['price_threshold'] is not None:
         text_input_placeholder = f"£{float(product_information['price_threshold']):.2f}"
-    new_threshold = st.text_input('Enter a new Threshold (£):',
+    st.markdown('<div style="position:relative;top:-1rem;"><p style="font-size:1.3rem; color: var(--orange); font-weight:600;">Update Price Alert</p></div>', unsafe_allow_html=True)
```
Comment (A-Faris) [inline Dashboard/pages/price.py]:
```suggestion
    st.markdown('<div style="position:relative;top:-1rem;"><p style="font-size:1.3rem;
    color: var(--orange); font-weight:600;">Update Price Alert</p></div>', unsafe_allow_html=True)
```

File: Dashboard/pages/price.py [hunk 8]
```diff
@@ -62,16 +67,21 @@ def change_threshold_in_db(
...
+
+    new_threshold = st.text_input('**Alert me when this product is below:** **:orange[*]**' ,
                                     placeholder=text_input_placeholder,
-                                    help='Can be left empty to remove threshold!',
                                     key=f'new_threshold_value_{product_information['product_id']}')
-    st.write('')
-    if st.button('Update Price Alert',
-                 key=f'new_threshold_submit_{product_information['product_id']}'):
+    st.markdown('<span style="font-size:0.8rem; position:relative; top:-1rem;">**:orange[*]** Leave this empty to be alerted on all price decreases</span>', unsafe_allow_html=True)
```
Comment (A-Faris) [inline Dashboard/pages/price.py]:
```suggestion
    st.markdown('<span style="font-size:0.8rem; position:relative; top:-1rem;">**:orange[*]**
    Leave this empty to be alerted on all price decreases</span>', unsafe_allow_html=True)
```

File: Dashboard/pages/price.py [hunk 9]
```diff
@@ -62,16 +67,21 @@ def change_threshold_in_db(
...
+    col1, col2 = st.columns([5,3])
+    with col1:
+        price_alert_update = st.button('Update Price Alert', key=f'new_threshold_submit_{product_information['product_id']}')
```
Comment (A-Faris) [inline Dashboard/pages/price.py]:
```suggestion
        price_alert_update = st.button('Update Price Alert',
        key=f'new_threshold_submit_{product_information['product_id']}')
```

File: Dashboard/pages/price.py [hunk 10]
```diff
@@ -191,18 +192,18 @@ def price_tracker_page(conn: connection) -> None:
...
         for website in websites:
             st.write('')
-            with st.expander(f'**{website}**'.title()):
+            with st.expander(f'**{website}**'):
                 for product in subscribed_to_products:
                     if product['website_name'] == website:
                         display_subscribed_product(conn, product)
     else:
-        st.warning('You are not subscribed to any products, please subscribe above!')
+        st.warning('You are not subscribed to track any product, please subscribe to track price changes.')
```
Comment (A-Faris) [inline Dashboard/pages/price.py]:
```suggestion
        st.warning('You are not subscribed to track any product, \
        please subscribe to track price changes.')
```

**Review by A-Faris — DISMISSED** [review]
It's great, it's just that lines are too long but you can ignore if you want

File: Dashboard/navigation.py (hunk 4 above)
Comment (BerkayDur) [reply Dashboard/navigation.py:25]: That's a good spot. I will create a new pull request to update the Icons.

**Review by BerkayDur — COMMENTED** [review]
Made the necessary changes!

File: Dashboard/pages/price.py (hunk 1 above)
Comment (BerkayDur) [reply Dashboard/pages/price.py]: Good Spot!

File: Dashboard/Dockerfile (hunk 3 above)
Comment (BerkayDur) [reply Dashboard/Dockerfile:18]: I'm going to create a new pull request for this! Thank you for notifying me!

**Review by A-Faris — APPROVED** [review]
Great

**Pull request merged**